src/Libraries/RevitNodes/Elements/Views/Sheet.cs

```diff
@@ -506,6 +521,110 @@ public static Sheet ByNameNumberTitleBlock(string sheetName, string sheetNumber,
             return new Sheet(sheetName, sheetNumber, titleBlockFamilyType.InternalFamilySymbol);
         }
 
+        /// <summary>
+        /// Duplicates A Sheet. 
+        /// </summary>
+        /// <param name="sheet">The Sheet to be Duplicated</param>
+        /// <param name="viewDuplicateOption">Enter View Duplicate Option: 0 = Duplicate. 1 = AsDependent. 2 = WithDetailing.</param>
+        /// <param name="prefix"></param>
+        /// <param name="suffix"></param>
+        /// <returns></returns>
+        public static Sheet DuplicateSheet(Sheet sheet, int viewDuplicateOption = 0, string prefix = "", string suffix = "")
+        {            
+            if (sheet == null)
+                throw new ArgumentNullException(nameof(sheet));
+
+            ViewDuplicateOption Option = 0;
+            switch (viewDuplicateOption)
+            {
+                case 0:
+                    Option = ViewDuplicateOption.Duplicate;
+                    break;
+                case 1:
+                    Option = ViewDuplicateOption.AsDependent;
+                    break;
+                case 2:
+                    Option = ViewDuplicateOption.WithDetailing;
+                    break;
+                default:
+                    throw new ArgumentException(Properties.Resources.ViewDuplicateOptionOutofRange);
+            }
+
+            if (String.IsNullOrEmpty(prefix) && String.IsNullOrEmpty(suffix))
+            {
+                throw new ArgumentNullException(Properties.Resources.SheetDuplicateNeedFix);
+            }
+
+            Sheet newSheet = null;
+
+            try
+            {
+                RevitServices.Transactions.TransactionManager.Instance.EnsureInTransaction(Application.Document.Current.InternalDocument);                
+
+                var oldElements = ElementBinder.GetElementsFromTrace<Autodesk.Revit.DB.Element>(Document);
+                List<ElementId> elementIds = new List<ElementId>();
+
+                if (oldElements != null)
+                {
+                    foreach (var element in oldElements)
+                    {
+                        elementIds.Add(element.Id);
+                    }
+                    Autodesk.Revit.UI.UIDocument uIDocument = new Autodesk.Revit.UI.UIDocument(Document);
+                    var openedViews = uIDocument.GetOpenUIViews().ToList();
+                    var shouldClosedViews = openedViews.FindAll(x => elementIds.Contains(x.ViewId));
+                    if (shouldClosedViews.Count > 0)
+                    {
+                        foreach (var v in shouldClosedViews)
+                            v.Close();
```

Comment:
To check if the view is the last view, if yes, delete those views that all ready closed.

Reply:
So delete all view but last one, do we need to throw exception if there is a last view?